c++/src/kj/compat/http.c++

```diff
@@ -2362,64 +2359,72 @@ public:
   }
 
   Promise<void> write(const void* buffer, size_t size) override {
-    if (size == 0) return kj::READY_NOW;  // can't encode zero-size chunk since it indicates EOF.
+    if (size == 0) co_return;  // can't encode zero-size chunk since it indicates EOF.
+
+    // The size of a hex-encoded uint64_t is 16 bytes. The CRLF and NUL terminator contribute an
+    // additional 3 bytes, for a total required space of 19.
+    kj::FixedArray<char, 19> headerSpace;
+    auto header = strPreallocated(headerSpace, kj::hex(size), "\r\n"_kj);
+
+    auto partCount = 4;
+    constexpr auto MAX_IN_FRAME = 256;
+    KJ_STACK_ARRAY(ArrayPtr<const byte>, parts, partCount, MAX_IN_FRAME, MAX_IN_FRAME);
```

Comment:
Can just be `ArrayPtr<const byte> parts[3];`

Reply:
I wonder what the hell I was thinking here ...